code/controllers/failsafe.dm

```diff
@@ -106,6 +115,41 @@ GLOBAL_REAL(Failsafe, /datum/controller/failsafe)
 			defcon = 5
 			sleep(initial(processing_interval))
 
+//Emergency loop used when Master got deleted or the main loop exited while Defcon == 0
+//Loop is driven externally so runtimes only cancel the current recovery attempt
+/datum/controller/failsafe/proc/emergency_loop()
+	//The code in this proc should be kept as simple as possible, anything complicated like to_chat might rely on master existing and runtime
+	//The goal should always be to get a new Master up and running before anything else
+	. = -1
+	switch (defcon) //The lower defcon goes the harder we try to fix the MC
+		if (4 to 5) //Try to normally recreate the MC two times
+			. = Recreate_MC()
+		if (3) //Delete the old MC first so we don't transfer any info, in case that caused any issues
+			del(Master)
+			. = Recreate_MC()
+		if (2) //Recreate all SSs which will still cause data survive due to Recover(), the new Master will then find and take them from global.vars
+			del(Master)
+			var/list/subsytem_types = subtypesof(/datum/controller/subsystem)
+			sortTim(subsytem_types, /proc/cmp_subsystem_init)
+			for(var/I in subsytem_types)
+				new I
+			. = Recreate_MC()
+		if (1) //Delete all existing SS to basically start over
+			del(Master)
+			for(var/global_var in global.vars)
+				if (istype(global.vars[global_var], /datum/controller/subsystem))
+					del(global.vars[global_var])
+			. = Recreate_MC()
+
+	if (. == 1) //We were able to create a new master
```

Comment:
The biggest issue i see with this logic is that recreate_mc will return 0 and not attempt to recreate the mc if it has been 'too soon' since it last attempted to recreate the mc, this should either send an argument to bypass this logic or handle it by not decrementing defcon in those cases.

Reply:
Fixed in latest commit
Will now only decrease defcon if Recreate_MC explicitly returns -1